lib/pennyworth/cli.rb

```diff
@@ -0,0 +1,339 @@
+        image_name = args.shift
+        VagrantCommand.setup_environment(@@settings.vagrant_dir)
+        if options[:url]
+          kiwi_dir = File.expand_path("~/.pennyworth/kiwi")
+          remote_url = options[:url]
+          opts = {
+            local: false,
+          }
+        else
+          if !Cli.settings.definitions_dir
+            STDERR.puts "You need to specify a definitions directory when not using --url."
+            exit 1
+          end
+          kiwi_dir = Cli.settings.kiwi_dir
+          opts = {
+            local: true
+          }
+        end
+        ImportBaseCommand.new(kiwi_dir, remote_url).execute(image_name, opts)
+      end
+    end
+
+    desc "Start virtual machine"
+    long_desc <<-LONGDESC
+      Start specified virtual machine
+    LONGDESC
+    arg_name "VM_NAME"
+    command :up do |c|
+      c.switch [:destroy], :default_value => false, :required => false, :negatable => false,
+        :desc => "Destroy vagrant instance(s) before starting them."
+      c.action do |global_options,options,args|
+        vm_name = args.shift
+        VagrantCommand.setup_environment(@@settings.vagrant_dir)
+        UpCommand.new.execute(vm_name, options)
+      end
+    end
+
+    desc "Stop virtual machine"
+    long_desc <<-LONGDESC
+      Stop specified virtual machine
+    LONGDESC
+    arg_name "VM_NAME"
+    command :down do |c|
+      c.action do |global_options,options,args|
+        vm_name = args.shift
+        VagrantCommand.setup_environment(@@settings.vagrant_dir)
+        DownCommand.new.execute(vm_name)
+      end
+    end
+
+    desc "Copy puplic SSH keys to target system"
+    long_desc <<-LONGDESC
+      Copy the public SSH keys to the host named IP-ADDRESS for easy root access
+    LONGDESC
+    arg_name "IP-ADDRESS"
+    command :copy_ssh_keys do |c|
+      c.flag [:password, :p], :type => String, :required => false,
+        :desc => "Password", :arg_name => "PASSWORD"
+      c.action do |global_options,options,args|
+        if !args.empty?
+          ip = args.shift
+        else
+          raise GLI::BadCommandLine.new("You need to provide the IP of the target system as argument.")
+        end
+
+        ImportSshKeysCommand.new.execute(ip, options)
+      end
+    end
+
+    desc "List available machines and images"
+    long_desc <<-LONGDESC
+      List virtual machines, base images, and related data
+    LONGDESC
+    command :list do |c|
+      c.action do |global_options,options,args|
+        VagrantCommand.setup_environment(@@settings.vagrant_dir)
+        ListCommand.new(Cli.settings.kiwi_dir).execute
+      end
+    end
+
+    desc "Boot custom image file"
+    long_desc <<-LONGDESC
+      Boot a custom image from the given file IMAGE in a VM not managed by
+      Vagrant. You can shut down an image booted this way with `shutdown`.
+    LONGDESC
+    arg_name "IMAGE"
+    command :boot do |c|
+      c.action do |global_options,options,args|
```

Comment:
Unused block argument - `global_options`. If it's necessary, use `_` or `_global_options` as an argument name to indicate that it won't be used.<br>Unused block argument - `options`. If it's necessary, use `_` or `_options` as an argument name to indicate that it won't be used.<br>Space missing after comma.